core/src/test/scala/org/apache/spark/jdbc/ConnectionConfTest.scala

```diff
@@ -0,0 +1,161 @@
+package org.apache.spark.jdbc
+
+import java.sql.{SQLException, DriverManager}
+
+import io.snappydata.SnappyFunSuite
+import io.snappydata.core.Data
+import org.apache.spark.{TaskContext, Logging}
+import org.scalatest.BeforeAndAfter
+
+import org.apache.spark.sql.SaveMode
+
+
+class ConnectionConfTest extends SnappyFunSuite with Logging with BeforeAndAfter {
+
+
+  test("test default conf") {
+    val conf = new ConnectionConfBuilder(snc).build()
+    assert(!conf.connProps.hikariCP)
+
+    val conn = ConnectionUtil.getPooledConnection("test default conf", conf)
+    assert(conn.getSchema != null)
+  }
+
+  test("test hikari conf") {
+    val conf = new ConnectionConfBuilder(snc).setPoolProvider("hikari").build()
+    assert(conf.connProps.hikariCP)
+
+    val conn = ConnectionUtil.getPooledConnection("test default conf", conf)
+    assert(conn.getSchema != null)
+  }
+
+
+  test("test Additional hikari conf") {
+    val conf = new ConnectionConfBuilder(snc).setPoolProvider("hikari").setPoolConf("maximumPoolSize", "50").build
+    assert(conf.connProps.hikariCP)
+    assert(conf.connProps.poolProps("maximumPoolSize") == "50")
+
+    val conn = ConnectionUtil.getPooledConnection("test default conf", conf)
+    assert(conn.getSchema != null)
+  }
+
+  test("test multiple hikari conf") {
+    val conf = new ConnectionConfBuilder(snc).setPoolProvider("hikari")
+        .setPoolConf("maximumPoolSize", "50")
+        .setPoolConf("minimumIdle", "5")
+        .build
+    assert(conf.connProps.hikariCP)
+    assert(conf.connProps.poolProps("maximumPoolSize") == "50")
+    assert(conf.connProps.poolProps("minimumIdle") == "5")
+
+    val conn = ConnectionUtil.getPooledConnection("test default conf", conf)
+    assert(conn.getSchema != null)
+  }
+
+  test("test serializibility") {
+    val data = Seq(Seq(1, 2, 3), Seq(7, 8, 9), Seq(9, 2, 3), Seq(4, 2, 3), Seq(5, 6, 7))
+    val rdd = sc.parallelize(data, data.length).map(s => new Data(s(0), s(1), s(2)))
+    val dataDF = snc.createDataFrame(rdd)
+
+    dataDF.write.format("row").mode(SaveMode.Append).saveAsTable("MY_SCHEMA.MY_TABLE")
+
+    val conf = new ConnectionConfBuilder(snc).build()
+
+    rdd.foreachPartition(d => {
```

Comment:
Sure will add one. I thought all example jobs get executed through quickStart. Apparently not.